Mojang icon (simple-icons#1708)

reginareynolds opened the pull request:
<!--
Before opening your pull request, have a quick look at our contribution guidelines: https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md
-->

**Issue:**


### Checklist
  - [X] I updated the JSON data in `_data/simple-icons.json`
  - [X] I optimized the icon with SVGO or SVGOMG
  - [X] The SVG `viewbox` is `0 0 24 24`

### Description
<!--
Anything relevant, for example:
  - Why did you pick the hex value?
  - Did you manually vectorize the logo?
  - Have you used multiple sources?
  - etc.
-->
The hex value is the color value of the logo. I checked the Mojang website for any design documents, but the official source linked in simple-icons#1708 was only about the terms and conditions of using the logo.

birjolaxew commented:
It actually looks like the header at [mojang.com](https://www.mojang.com/) contains an SVG; the website itself is no longer used, but it looks like the logo is the current one.

reginareynolds commented:
@birjolaxew You're right, the logo off of [mojang.com](mojang.com) is perfectly square while the previous version is ever so slightly rectangular. The shades of red are also slightly different. I updated the image source and the hex value to reflect the new source .svg. I've also updated the icon based off of the new image source

birjolaxew commented:
:+1: Great to hear! Thanks for the speedy update and the work put into this PR, we appreciate it a lot.
